File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_amount_kind/DisplayAmountKindStrategyFactoryTest.java
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_amount_kind;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

import coolpharaoh.tee.speicher.tea.timer.core.tea.AmountKind;

class DisplayAmountKindStrategyFactoryTest {

    @Test
    void getDisplayAmountKindGram() {
        final DisplayAmountKindStrategy displayAmountKindStrategy = DisplayAmountKindFactory.get(AmountKind.GRAM, null);
        assertThat(displayAmountKindStrategy).isInstanceOf(DisplayAmountKindStrategyGram.class);
    }

    @Test
    void getDisplayAmountKindTeaBag() {
        final DisplayAmountKindStrategy displayAmountKindStrategy = DisplayAmountKindFactory.get(AmountKind.TEA_BAG, null);
        assertThat(displayAmountKindStrategy).isInstanceOf(DisplayAmountKindStrategyTeaBag.class);
    }

    @Test
    void getDisplayAmountKindTeaSpoon() {
        final DisplayAmountKindStrategy displayAmountKindStrategy = DisplayAmountKindFactory.get(AmountKind.TEA_SPOON, null);
        assertThat(displayAmountKindStrategy).isInstanceOf(DisplayAmountKindStrategyTeaSpoon.class);
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_amount_kind/DisplayAmountKindStrategyFactoryTest.kt
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_amount_kind

import android.app.Application
import coolpharaoh.tee.speicher.tea.timer.core.tea.AmountKind
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class DisplayAmountKindStrategyFactoryTest {

    @Test
    fun getDisplayAmountKindGram() {
        val displayAmountKindStrategy = DisplayAmountKindFactory[AmountKind.GRAM, Application()]
        assertThat(displayAmountKindStrategy).isInstanceOf(DisplayAmountKindStrategyGram::class.java)
    }

    @Test
    fun getDisplayAmountKindTeaBag() {
        val displayAmountKindStrategy = DisplayAmountKindFactory[AmountKind.TEA_BAG, Application()]
        assertThat(displayAmountKindStrategy).isInstanceOf(DisplayAmountKindStrategyTeaBag::class.java)
    }

    @Test
    fun getDisplayAmountKindTeaSpoon() {
        val displayAmountKindStrategy = DisplayAmountKindFactory[AmountKind.TEA_SPOON, Application()]
        assertThat(displayAmountKindStrategy).isInstanceOf(DisplayAmountKindStrategyTeaSpoon::class.java)
    }
}